static/js/apps/nl_interface/app_state.ts

```diff
@@ -0,0 +1,382 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Application state management.
+ *
+ * Documentation: https://easy-peasy.vercel.app/
+ */
+
+import axios from "axios";
+import {
+  Action,
+  action,
+  createStore,
+  createTypedHooks,
+  Thunk,
+  thunk,
+} from "easy-peasy";
+import _ from "lodash";
+import Papa from "papaparse";
+
+import {
+  NL_DETECTOR_VALS,
+  NL_INDEX_VALS,
+  NL_PLACE_DETECTOR_VALS,
+  NL_URL_PARAMS,
+} from "../../constants/app/nl_interface_constants";
+import { SearchResult } from "../../types/app/nl_interface_types";
+import { stringifyFn } from "../../utils/axios";
+import { getUrlToken, getUrlTokenOrDefault } from "../../utils/url_utils";
+
+/**
+ * Sample NL query
+ */
+export interface SampleQuery {
+  topic: string;
+  query: string;
+  // "Works as intended"
+  wai: boolean;
+}
+
+export interface NLQueryType {
+  chartData?: SearchResult;
+  context?: any; // Summarized context
+  debugData?: any;
+  errorMsg?: any;
+  feedbackGiven?: boolean;
+  id: string;
+  isLoading: boolean;
+  nlQueryContextId: string;
+  nlQueryIdHistory: string[];
+  query: string; // Query text
+  response?: any; // Response from NL API
+  timestamp: number; // Query epoch time in ms
+  took?: number; // Query response time in ms
+}
+
+export interface NLQueryContextType {
+  id: string;
+  name: string;
+  nlQueryIds: string[];
+}
+
+interface NLAppConfig {
+  autoRun: boolean;
+  currentNlQueryContextId: string;
+  delayDisabled: boolean;
+  demoMode: boolean; // Show feedback buttons
```

Comment:
I think demoMode hides the feedback buttons right?

Reply:
Fixed!